src/style-spec/validate/validate_expression.js

```diff
@@ -1,25 +1,15 @@
 
 const ValidationError = require('../error/validation_error');
 const {createExpression} = require('../expression');
-const {getExpectedType, getDefaultValue} = require('../expression');
 const unbundle = require('../util/unbundle_jsonlint');
 
 module.exports = function validateExpression(options) {
-    const defaultValue = options.key.endsWith('heatmap-color') ? [0, 0, 0, 0] :
-        getDefaultValue(options.valueSpec);
-    const expression = createExpression(
-        unbundle.deep(options.value), {
-            context: options.expressionContext,
-            expectedType: getExpectedType(options.valueSpec),
-            defaultValue
-        });
-
+    const expression = createExpression(unbundle.deep(options.value), options.valueSpec, 'property');
```

Comment:
Shouldn't `'property'` be part of `options`, since this function will be used for validating filters, too?